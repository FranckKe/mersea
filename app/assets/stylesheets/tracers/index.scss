.off-canvas-wrapper, .off-canvas-wrapper-inner {
height: 100%;
}

.report-link {
    color: red;
    padding-top: 1rem;
}

.traceur-title {
    text-align: center;
    padding-top: 1rem;
}

.tracers-flex-container {
    padding-bottom: 4rem;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    flex-wrap: wrap;
    background-color: inherit;
    height: 100%;

    .card {
        flex-basis: 150px;
        flex-shrink: 1;
        flex-grow: 1;
        margin: .5rem;
        overflow: hidden;
        box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);

        .title {
            line-height: 3rem;
            font-size: 1.5rem;
            font-weight: 300;
        }
        .content {
            padding: .7rem;
            font-weight: 300;
            border-radius: 0 0 2px 2px;
        }
        p {
            margin: 0;
        }
        .action {
            border-top: 1px solid rgba(160, 160, 160, 0.2);
            padding: .7rem;
        }
        a {
            display: block;
            margin-right: 1.3rem;
            transition: color 0.3s ease;
            text-transform: uppercase;
            text-decoration: none;
            font-size: .8rem;
            word-wrap: break-word;
        }
        .image {
            position: relative;
            width: 100%;
            .title {
                position: absolute;
                bottom: 0;
                left: 0;
                padding: 1.3rem;
                color: #fff;
            }
            img {
                border-radius: 2px 2px 0 0;
            }
        }
    }
}