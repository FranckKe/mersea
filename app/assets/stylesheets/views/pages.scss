main {
  height: 100%;
}

.map {
  width: 100%;
  height: 100%;

  .leaflet-bottom.leaflet-right .leaflet-control-layers {
    max-height: calc(100vh - 55px - 74px - 10px - 32px - 5px); // Menu - Zoom Buttons - Bottom Margin - Footer - 5px
    overflow-y: scroll;
  }
}

.leaflet-marker-icon {
  max-width: 64px;
  max-height: 64px;
}

.article {
  width: 100%;
  max-width: 750px;
  margin: auto;
  padding-top: 25px;
  padding-right: 25px;
  padding-left: 25px;

  h1 {
    text-align: center;
  }
}
