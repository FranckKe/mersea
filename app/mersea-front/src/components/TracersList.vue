<template>
  <!--<table class="table is-fullwidth is-striped is-narrow is-hoverable is-fullwidth">
    <tbody>
      <tr :key="idx" v-for="(tracer, idx) in tracersData()">
        <td>{{ tracer.name }}</td>
      </tr>
    </tbody>
  </table>-->
  <div class="container">
    <b-field grouped group-multiline>
      <b-select v-model="perPage">
        <option value="5">5 per page</option>
        <option value="10">10 per page</option>
        <option value="15">15 per page</option>
        <option value="20">20 per page</option>
      </b-select>
    </b-field>
    <b-table
      :data="tracersData()"
      :paginated="true"
      :per-page="perPage"
      :current-page.sync="currentPage"
      :pagination-simple="false"
      :default-sort-direction="defaultSortDirection"
      default-sort="name">
      <template slot-scope="props">
        <b-table-column field="name" label="Name" width="40" sortable numeric>
          {{ props.row.name }}
        </b-table-column>
      </template>
    </b-table>
  </div>
</template>

<script>
export default {
  name: "tracers-list",
  props: ['tracers'],
  data() {
    return {
      tracersData: this.tracers,
      defaultSortDirection: 'asc',
      currentPage: 1,
      perPage: 10
    }
  }
}
</script>
