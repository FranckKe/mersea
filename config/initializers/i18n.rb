Rails.application.config.i18n.available_locales = %i[fr en es]
Rails.application.config.i18n.default_locale = 'fr'
