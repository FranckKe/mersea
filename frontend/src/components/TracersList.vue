<template>
  <div>
    <b-field grouped group-multiline>
      <b-select v-model="perPage">
        <option value="5">5 {{ $t('per_page') }}</option>
        <option value="10">10 {{ $t('per_page') }}</option>
        <option value="15">15 {{ $t('per_page') }}</option>
        <option value="20">20 {{ $t('per_page') }}</option>
      </b-select>
    </b-field>
    <b-table
      :data="tracers"
      :paginated="true"
      :per-page="perPage"
      :current-page.sync="currentPage"
      :pagination-simple="false"
      :default-sort-direction="defaultSortDirection"
      :opened-detailed="defaultOpenedDetails"
      default-sort="name"
      detailed
      detail-key="id">
      <template slot-scope="props">
        <b-table-column field="name" v-bind:label="$t('name')" sortable>
          {{ props.row.name }}
        </b-table-column>
        <b-table-column field="name" v-bind:label="$t('origin')" sortable>
          {{ props.row.origin }}
        </b-table-column>
        <b-table-column field="kind" v-bind:label="$t('type')" sortable>
          {{ props.row.kind }}
        </b-table-column>
        <b-table-column field="created_at" v-bind:label="$t('created_at')" sortable>
          {{ props.row.created_at | formatDate }}
        </b-table-column>
      </template>
      <template slot="detail" slot-scope="props">
        <article class="media">
          <figure class="media-left">
            <p class="image is-64x64">
              <img :src="`${apiUrl}${props.row.photo}`">
            </p>
          </figure>
          <div class="media-content">
            <div class="content">
              <p class="tracer-description">{{ props.row.description }}</p>
            </div>
          </div>
        </article>
      </template>
      <template slot="empty">
        <section class="section">
          <div class="content has-text-grey has-text-centered">
            <p>
              <b-icon
                icon="emoticon-sad"
                size="is-large">
              </b-icon>
            </p>
            <p>{{ $t('nothing') }}</p>
          </div>
        </section>
      </template>
    </b-table>
  </div>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'
const { mapState, mapGetters, mapMutations } = createNamespacedHelpers(
  'tracers'
)

export default {
  name: 'tracers-list',
  props: ['tracers'],
  data() {
    return {
      apiUrl: this.$apiUrl,
      defaultSortDirection: 'asc',
      currentPage: 1,
      defaultOpenedDetails: [1]
    }
  },
  methods: {
    ...mapMutations(['setPerPage']),
    ...mapGetters(['getPerPage'])
  },
  computed: {
    ...mapState({
      perPage: state => state.perPage
    }),
    // Two-way Computed Property
    perPage: {
      set(value) {
        this.setPerPage(value)
      },
      get() {
        return this.getPerPage()
      }
    }
  }
}
</script>


<style>
.tracer-description {
  max-width: 700px;
}
</style>

<i18n>
{
  "en": {
    "created_at": "Created at",
    "name": "Name",
    "nothing": "No data",
    "origin": "Origin",
    "per_page": "per page",
    "type": "Type"
  },
  "fr": {
    "created_at": "Créé le",
    "name": "Nom",
    "nothing": "Pas de données",
    "origin": "Origine",
    "per_page": "par page",
    "type": "Type"
  },
  "es": {
    "created_at": "Creado en",
    "name": "Apellido",
    "nothing": "Sin datos",
    "origin": "Origen",
    "per_page": "por página",
    "type": "Tipo"
  }
}
</i18n>
