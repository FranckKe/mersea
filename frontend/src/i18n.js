import Vue from 'vue'
import VueI18n from 'vue-i18n'

Vue.use(VueI18n)

function loadLocaleMessages() {
  const locales = require.context(
    './locales',
    true,
    /[A-Za-z0-9-_,\s]+\.json$/i
  )
  const messages = {}
  locales.keys().forEach(key => {
    const matched = key.match(/([a-z0-9]+)\./i)
    if (matched && matched.length > 1) {
      const locale = matched[1]
      messages[locale] = locales(key)
    }
  })
  return messages
}

// Detect browser language and strip the "region" language
// en-GB => en
let browserI18n = navigator.language
if (navigator.language.indexOf('-') >= 0)
  browserI18n = navigator.language.substring(0, navigator.language.indexOf('-'))

// Default to english if browser language is not supported
let initializedLocal = ['en', 'es', 'fr'].includes(browserI18n)
  ? browserI18n
  : 'en'

export default new VueI18n({
  locale: initializedLocal,
  fallbackLocale: 'en',
  messages: loadLocaleMessages()
})
