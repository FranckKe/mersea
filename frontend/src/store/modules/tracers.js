import axios from 'axios'

const api = axios.create({
  baseURL: process.env.VUE_APP_API_URL,
  headers: { Accept: 'application/json', 'Content-Type': 'application/json' }
})

const state = {
  tracers: [],
  filteredTracers: [],
  loading: true,
  success: false,
  errors: [],
  displayFormat: 'grid',
  perPage: '10'
}

const getters = {
  getFilteredTracers: state => {
    return state.filteredTracers
  },
  getLoading: state => {
    return state.loading
  },
  getError: state => {
    return state.errors
  },
  getSuccess: state => {
    return state.success
  },
  getTracers: state => {
    return state.tracers
  },
  getPerPage: state => {
    return state.perPage
  },
  getTracerById: state => tracerId =>
    state.tracers.filter(tracer => tracerId === tracer.id)[0]
}

const mutations = {
  setTracers: (state, { tracers }) => {
    state.tracers = tracers
  },
  setLoading: (state, loading) => {
    state.loading = loading
  },
  setSuccess: state => {
    state.success = true
    state.loading = false
    state.errors = []
  },
  setError: (state, { errors }) => {
    state.errors = errors
    state.loading = false
  },
  setDisplayFormat: (state, format) => {
    state.displayFormat = format
  },
  setPerPage: (state, perPage) => {
    state.perPage = perPage
  },
  setFilteredTracers: (state, filteredTracers) => {
    state.filteredTracers = filteredTracers
  }
}

const actions = {
  async loadTracers({ commit }) {
    try {
      commit('setLoading', true)

      const tracersRes = await api.get(`/tracers`)
      const tracers = tracersRes.data
      commit('setTracers', { tracers })
      commit('setFilteredTracers', tracers.map(t => t.id))
      commit('setSuccess')
    } catch (e) {
      let errors = [e.message]
      commit('setError', { errors })
    }
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}
