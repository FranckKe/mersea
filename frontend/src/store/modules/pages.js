import axios from 'axios'
import i18n from '../../i18n'
import slugify from 'slugify'

const api = axios.create({
  baseURL: process.env.VUE_APP_API_URL,
  headers: { Accept: 'application/json', 'Content-Type': 'application/json' }
})

const state = {
  pages: [],
  loading: true,
  success: false,
  errors: []
}
const getters = {
  getLoading: state => {
    return state.loading
  },
  getError: state => {
    return state.errors
  },
  getSuccess: state => {
    return state.success
  },
  getPageContent: state => {
    return pageInfo => {
      return (
        state.pages.filter(page => page.slug === pageInfo.pageName)[0] || {}
      ).content
    }
  },
  getCategories: state => {
    return [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
  },
  getAllPagesByCategory: state => {
    let categories = [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
    return categories.map(function(category) {
      return {
        category: slugify(category),
        pagesName: state.pages
          .filter(
            page => page.category === category && page.language === i18n.locale
          )
          .map(page => {
            return { raw: page.name, slug: page.slug }
          })
      }
    })
  },
  getPagesByCategory(state) {
    return category => {
      return state.pages
        .filter(
          page => page.category === category && page.language === i18n.locale
        )
        .map(page => {
          return { raw: page.name, slug: page.slug }
        })
    }
  }
}
const mutations = {
  setPages: (state, { pages }) => {
    state.pages = pages
  },
  setSuccess: state => {
    state.success = true
    state.loading = false
    state.errors = []
  },
  setError: (state, { errors }) => {
    state.errors = errors
    state.loading = false
  }
}
const actions = {
  async loadPages({ commit }) {
    try {
      const pagesData = await api.get(`/pages`)
      const pages = pagesData.data
      commit('setPages', { pages })
      commit('setSuccess')
    } catch (e) {
      let errors = [e.message]
      commit('setError', { errors })
    }
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}
