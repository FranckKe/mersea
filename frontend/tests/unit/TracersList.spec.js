import VueTestUtils from '@vue/test-utils'
import TracersList from '@/components/TracersList.vue'
import Vuex from 'vuex'
import pages from '../../src/store/modules/pages'
import tracers from '../../src/store/modules/tracers'
import Buefy from 'buefy'
import VueRouter from 'vue-router'
import i18n from '../../src/i18n'
import moment from 'moment'
import tracersData from './data/tracers'

VueTestUtils.config.mocks.$t = key => key
VueTestUtils.config.mocks.$tc = (key, count) => (key, count)

const localVue = VueTestUtils.createLocalVue()
localVue.use(Buefy)
localVue.use(Vuex)
localVue.use(VueRouter)
localVue.filter('formatDate', value =>
  moment(String(value)).format('MM/DD/YYYY')
)

const router = new VueRouter()

describe('Page.vue', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        pages: pages,
        tracers: tracers
      }
    })
  })

  it('matches snapshot', () => {
    const wrapper = VueTestUtils.shallowMount(TracersList, {
      localVue,
      store,
      router,
      i18n,
      propsData: {
        tracers: tracersData
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
