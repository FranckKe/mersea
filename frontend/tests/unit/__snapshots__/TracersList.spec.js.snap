// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page.vue matches snapshot 1`] = `
<div>
  <bfield-stub grouped="true" groupmultiline="true" addons="true">
    <bselect-stub value="10">
      <option value="5">5 per_page</option>
      <option value="10">10 per_page</option>
      <option value="15">15 per_page</option>
      <option value="20">20 per_page</option>
    </bselect-stub>
  </bfield-stub>
  <btable-stub data="[object Object],[object Object],[object Object],[object Object]" columns="" detailed="true" isrowcheckable="function (){return!0}" checkedrows="" mobilecards="true" defaultsort="name" defaultsortdirection="asc" paginated="true"
    currentpage="1" perpage="10" rowclass="function (){return&quot;&quot;}" openeddetailed="1" hasdetailedvisible="function (){return!0}" detailkey="id" total="0"></btable-stub>
</div>
`;
